Client/Frontend/Browser/OpenSearch.swift

```diff
@@ -164,39 +166,41 @@ class OpenSearchParser {
             return nil
         }
 
-        var uiImage: UIImage?
-        var imageIndexers = docIndexer["Image"].all
+        let imageIndexers = docIndexer["Image"].all
+        var largestImage = 0
+        var largestImageElement: XMLElement?
 
+        // TODO: For now, just use the largest icon.
         for imageIndexer in imageIndexers {
-            // TODO: For now, we only look for 16x16 search icons.
-            let imageWidth = imageIndexer.element?.attributes["width"]
-            let imageHeight = imageIndexer.element?.attributes["height"]
-            if imageWidth?.toInt() != 16 || imageHeight?.toInt() != 16 {
-                continue
-            }
+            let imageWidth = imageIndexer.element?.attributes["width"]?.toInt()
+            let imageHeight = imageIndexer.element?.attributes["height"]?.toInt()
 
-            if imageIndexer.element?.text == nil {
+            // Only accept square images.
+            if imageWidth != imageHeight {
```

Comment:
Both `imageWidth` and `imageHeight` are optionals, so this check may not do what you intended. Like, it will pass if both are `nil`.